File: services/typescript/fastify/src/task.ts
import * as z from "zod";
import { Operation } from "./operation.js";

export const Task = z.object({
  secret: z.string(),
  operation: z.enum(Operation)
});

export type Task = z.infer<typeof Task>;

export const PatchedTask = z.object({
  operation: z.enum(Operation)
});

export type PatchedTask = z.infer<typeof PatchedTask>;

File: shared-code/rust/app/src/task.rs
use crate::operation::Operation;
use serde::{Deserialize, Serialize};

/// The payload expected by a web-framework from a POST/PUT request
#[derive(Debug, Deserialize, Serialize, Clone)]
pub struct Task {
    /// Arbitrary data to differentiate payloads.
    pub secret: String,

    /// The operation to perform on the payload.
    pub operation: Operation,
}

/// The payload expected by a web-framework from a PATCH request
#[derive(Debug, Deserialize, Serialize, Clone)]
pub struct PatchedTask {
    pub operation: Operation,
}
